src/main/java/thePackmaster/cardmodifiers/frostpack/FrozenMod.java

```diff
@@ -0,0 +1,93 @@
+package thePackmaster.cardmodifiers.frostpack;
+
+import basemod.abstracts.AbstractCardModifier;
+import basemod.helpers.CardModifierManager;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.util.TexLoader;
+import thePackmaster.util.Wiz;
+
+public class FrozenMod extends AbstractCardModifier {
+    public static String ID = SpireAnniversary5Mod.makeID("FrozenMod");
+
+    private int originalCost;
+    private boolean hadRetain = false;
+    private static final Texture tex = TexLoader.getTexture(SpireAnniversary5Mod.modID + "Resources/images/ui/frozenOverlay.png");
+
+
+    @Override
+    public boolean canPlayCard(AbstractCard card) {
+        return false;
+    }
+
+    @Override
+    public void onInitialApplication(AbstractCard card) {
+
+        CardCrawlGame.sound.play("ORB_FROST_CHANNEL", 0.1F);
+        originalCost = card.cost;
+        card.modifyCostForCombat(1);
+        if (card.selfRetain) hadRetain = true;
+        card.selfRetain = true;
+        card.tags.add(SpireAnniversary5Mod.FROZEN);
+    }
+
+    public String identifier(AbstractCard card) {
+        return ID;
+    }
+
+
+    @Override
+    public void onRemove(AbstractCard card) {
+        card.tags.remove(SpireAnniversary5Mod.FROZEN);
+
+        //Adds another modifier which hands over the original cost of the card when Frozen, so the card can revert back to its original cost when played.
+        CardModifierManager.addModifier(card, new RevertCostWhenPlayedMod(originalCost));
+        Wiz.atb(new AbstractGameAction() {
+            @Override
+            public void update() {
+                this.isDone = true;
+                if (!hadRetain) {
+                    card.selfRetain = false;
+                }
+            }
+        });
+    }
+
+    @Override
+    public void onRender(AbstractCard card, SpriteBatch sb) {
```

Comment:
Yep, verified this.... no idea why though.  Maybe a cardmodifiers bug?

Reply:
A CardModifiers bug is a reasonable guess, though the patch that calls this method is simple and straightforward: https://github.com/daviscook477/BaseMod/blob/1a8a344ecfde07613d29f39ff52b380cf50196f9/mod/src/main/java/basemod/patches/com/megacrit/cardcrawl/cards/AbstractCard/CardModifierPatches.java#L321

Just to be sure, while testing locally, I verified that the `onRender` method is getting called when I freeze a Tactician by adding the following code. Sure enough, I got a window full of that log statement when I created and froze a Tactician, so it's getting called. So somehow, the `sb.draw` call has no effect -- it's being cancelled out or overridden later
```
        if (card.cardID.equals(Tactician.ID)) {
            SpireAnniversary5Mod.logger.info("In onRender for Tactician");
        }
```

Also, I used the Pyromaniac power from the Rimworld pack to apply `OnFireModifier` to a Tactician, and the extra icon that adds _does_ render. So whatever the issue is, it's something that is only a problem when you're immediately trying to render things (extra icons queues up the rendering for later, from what I understand).
![image](https://user-images.githubusercontent.com/62083413/224470624-5277c61f-3cc1-40ae-9b5b-87f2229f85dd.png)

So it looks to me like there's some interaction we don't expect that's causing this -- something that causes `SpriteBatch.draw` to have a different result for some cards (seems to be Unplayable cards) than for others. That's some information, but it's still tricky to narrow it down any further. It could be that there's something we're missing in BaseMod's CardModifier code that explains this. Or it could be that one of the 100+ other patches in Packmaster is messing this up. Or even that there's just something a bit different about rendering Unplayable cards deep within the base game code.

You might be able to tell the difference by reproducing `FrozenMod` in a clean non-Packmaster shell... that's a time consuming step to take, so probably worth some other debugging first, but might end up needing to do something like that. I just spent a bit looking through code and trying things out locally to test some hypotheses and didn't find very much (except that it's not the fault of the OccultPatch, blend functions are set the same for both playable and unplayable cards, and calling sb.flush() doesn't change anything).

Reply:
I suppose it's worth asking in Discord if anyone has run into this before -- even if the odds might be low, no harm in trying. You never know when somehow might pop up with some unexpected insight.

Reply:
Perhaps we can design our way around this and say "Unplayable card cannot be Frozen".  I already exclude Frozen cards from being freezeable though it's not specified that is true anywhere.  For design purposes, it would prevent freezing of things like Regret over and over again from Extended Stall, at the cost of not being able to freeze a Tactician throught he same method, but players are more likely to be preferable to that.  But the problem there would be figuring out a good way to catch all unplayables since thats not a clean variable declaration anywhere.

Reply:
And one last thought on this before I'm out of the time I have to dig into this -- even if we can't work out what's going on here, maybe the way ExtraIcons works gives us a potential workaround. We could change FrozenMod to defer rendering in the same way as ExtraIcons. In other words, instead of calling `sb.draw` immediately, we patch things to do the rendering downstream. 
Looks to me like the way ExtraIcons does it is with the following patch: https://github.com/kiooeht/StSLib/blob/e9679f117c9bdc77e592b50c385a544f70f4a793/src/main/java/com/evacipated/cardcrawl/mod/stslib/patches/ExtraIconsPatch.java

It's not clear to me that this would solve the problem (can't really be confident about that when I don't _understand_ the problem), but it might, and we're at the point where our path for further investigation is unclear enough that it's worth considering if we can dodge the problem somehow.

Reply:
I do think my idea of doing things like ExtraIcons is worth trying -- it has a chance of working.

If that doesn't work, and there's no quick breakthrough on why this is happening, I agree that the practical choice is making it so that Unplayable cards cannot be Frozen. The nice thing is that Alch already did the heavy lifting of writing a patch to figure out unplayable cards and you can use that by calling `OccultPatch.isUnplayable`. It's currently only used by one power (Gravity) from the Psychic pack, but it should be reliable and Frost pack can use it too.

Reply:
I ended up just going with Unplayables can't be frozen. I think that's the ultimately better path for the gameplay due to the existence of Extended Stall and clearly explains why Frozen cards can't be re-frozen

Reply:
Sounds good. I also ended up feeling like that was a pretty good choice anyway since so many Unplayable cards are curses and statuses you don't want to keep anyway.

Looks like everything is in good shape so I'll merge this in now.